### PR TITLE
python312Packages.pysmi: 0.3.4 -> 1.4.4; python312Packages.pysnmp: 4.4.2 -> 6.2.2

### PR DESCRIPTION
**Pull request by mweinelt**
Removes the lextudio forks, which now have become the canonical pysmi/pysnmp source.

https://github.com/lextudio/pysmi/compare/refs/tags/v0.3.4...v1.4.4
https://github.com/lextudio/pysnmp/blob/v6.2.2/CHANGES.rst

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 328047` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>blocksat-cli (python312Packages.blocksat-cli)</li>
    <li>blocksat-cli.dist (python312Packages.blocksat-cli.dist)</li>
    <li>home-assistant-component-tests.snmp</li>
    <li>routersploit</li>
    <li>routersploit.dist</li>
  </ul>
</details>
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>glances</li>
    <li>glances.dist</li>
    <li>home-assistant-component-tests.brother</li>
    <li>python311Packages.atenpdu</li>
    <li>python311Packages.atenpdu.dist</li>
    <li>python311Packages.blocksat-cli</li>
    <li>python311Packages.blocksat-cli.dist</li>
    <li>python311Packages.brother</li>
    <li>python311Packages.brother.dist</li>
    <li>python311Packages.patator</li>
    <li>python311Packages.patator.dist</li>
    <li>python311Packages.pysmi</li>
    <li>python311Packages.pysmi.dist</li>
    <li>python311Packages.pysnmp</li>
    <li>python311Packages.pysnmp.dist</li>
    <li>python312Packages.atenpdu</li>
    <li>python312Packages.atenpdu.dist</li>
    <li>python312Packages.brother</li>
    <li>python312Packages.brother.dist</li>
    <li>python312Packages.patator</li>
    <li>python312Packages.patator.dist</li>
    <li>python312Packages.pysmi</li>
    <li>python312Packages.pysmi.dist</li>
    <li>python312Packages.pysnmp</li>
    <li>python312Packages.pysnmp.dist</li>
  </ul>
</details>

**Comment by mweinelt** [comment]
Routersploit is blocked by the removal of the sync high-level API.

https://github.com/threat9/routersploit/issues/819

File: pkgs/development/python-modules/pysnmp/default.nix [hunk 1]
```diff
@@ -1,40 +1,74 @@
...
+  # tests
+  pytestCheckHook,
+  pytest-asyncio,
 }:
 
 buildPythonPackage rec {
   pname = "pysnmp";
-  version = "4.4.12";
-  format = "setuptools";
+  version = "6.2.2";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pysnmp/default.nix:21]:
```suggestion
  version = "6.2.5";
```

File: pkgs/servers/home-assistant/update-component-packages.py [hunk 2]
```diff
@@ -44,6 +44,7 @@
     "numpy": "numpy_1",
     "ollama-hass": "ollama",
     "paho-mqtt": "paho-mqtt_1",
+    "pysnmp-lextudio": "pysnmp",
```
Comment (dotlambda) [inline pkgs/servers/home-assistant/update-component-packages.py:47]: Not needed anymore: https://github.com/NixOS/nixpkgs/pull/338722

File: pkgs/servers/home-assistant/default.nix [hunk 3]
```diff
@@ -540,6 +540,8 @@ in python.pkgs.buildPythonApplication rec {
       src = ./patches/ffmpeg-path.patch;
       ffmpeg = "${lib.getBin ffmpeg-headless}/bin/ffmpeg";
     })
+
+    ./patches/pysnmp-compat.patch
```
Comment (dotlambda) [inline pkgs/servers/home-assistant/default.nix:544]: also no longer needed

**Pull request closed without merging**